launcher/common/launcher.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+# ------------------------------------------------------------------------------
+#  Copyright 2020 Forschungszentrum Jülich GmbH and Aix-Marseille Université
+# "Licensed to the Apache Software Foundation (ASF) under one or more contributor
+#  license agreements; and to You under the Apache License, Version 2.0. "
+#
+# Forschungszentrum Jülich
+#  Institute: Institute for Advanced Simulation (IAS)
+#    Section: Jülich Supercomputing Centre (JSC)
+#   Division: High Performance Computing in Neuroscience
+# Laboratory: Simulation Laboratory Neuroscience
+#       Team: Multiscale Simulation and Design
+#
+#   Date: 2021.02.16
+# ------------------------------------------------------------------------------
+import os
+import common
+import configurations_manager
+# import xml
+import json
+
+
+class Launcher:
+    """
+        Class representing the co-simulation launcher tool
+
+    Methods:
+    --------
+        run(args=None)
+            Entry point to the launcher and executes the main loop of the tool
+    """
+    # general members
+    __args = None
+    __configuration_manager = None
+    __logs_root_dir = None
+    __logger = None
+
+    # XML validators
+    __co_sim_parameters_validator = None
+    __co_sim_plan_validator = None
+
+    # dictionaries
+    __action_plan_parameters_dict = {}
+    __action_plan_variables_dict = {}
+    __parameters_parameters_dict = {}
+    __parameters_variables_dict = {}
+
+    def generate_parameters_json_file(self):
+        """
+            Dumps into the /path/to/co_sim/results/dir/filename.json file
+            the parameters gathered from the parameters XML file
+
+        :return:
+            JSON_FILE_ERROR: reporting error during the parameter JSON file
+            OK: parameter JSON file was generated properly
+        """
+
+        # TO BE DONE: exception management when the file cannot be created
+
+        results_dir = self.__configuration_manager.get_directory('results')
+        parameters_json_file = self.__parameters_parameters_dict['CO_SIM_PARAMETERS_JSON_FILENAME']
+
+        with open(results_dir+'/'+parameters_json_file, 'w') as json_output_file:
```

Comment:
It might be that I misunderstood it but ain't XML was chosen as a 'standard' instead of JSON or YAML?